### DISCUSSION TITLE
Question Re Spindle PWM

### DISCUSSION THREAD

**Discussion by rvalotta** [terjeio/grblHAL#334] [Q&A]
Is it possible to have 2 different spindle PWM outs?  1 for a spindle and 1 for a laser?  Where it switches based on $32 or something similar?    Thinking of adding a laser diode to the gantry so i can engrave my logo or something similar.

**Comment by terjeio** [comment]
> Is it possible to have 2 different spindle PWM outs? 1 for a spindle and 1 for a laser?

Yes.

> Where it switches based on $32 or something similar? 

Easiest would be to switch using a custom M-code (or $32), either to [control a relay](https://github.com/grblHAL/Templates/blob/master/my_plugin/probe%20select/my_plugin.c) or switch between PWM generators. Both can be done from a plugin. Here is [an example](https://github.com/grblHAL/Plugins_spindle/blob/cdaa91c8e68eb55270ebf0c1202cc72a24231e84/huanyang.c#L274-L353) for how switching by swapping function pointers is done by trapping $32.

If you switch often and have setting stored in flash then $32 will add to flash wear.
